apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/util.ts

```diff
@@ -1128,3 +1130,136 @@ export const getEntitiesByType: ImpureGraphFunction<
 export const anyUserInstantiator: EntityTypeInstantiatorSubject = {
   kind: "public",
 };
+
+export const getExistingUsersAndOrgs: ImpureGraphFunction<
+  {},
+  Promise<{ users: Entity[]; orgs: Entity[] }>
+> = async (context, authentication) => {
+  const users = await getEntities(context, authentication, {
+    query: {
+      filter: {
+        all: [
+          {
+            equal: [
+              { path: ["type", "baseUrl"] },
+              { parameter: systemEntityTypes.user.entityTypeBaseUrl },
+            ],
+          },
+        ],
+      },
+      graphResolveDepths: zeroedGraphResolveDepths,
+      includeDrafts: true,
+      temporalAxes: currentTimeInstantTemporalAxes,
+    },
+  }).then((subgraph) => getRoots(subgraph));
+
+  const orgs = await getEntities(context, authentication, {
+    query: {
+      filter: {
+        all: [
+          {
+            equal: [
+              { path: ["type", "baseUrl"] },
+              { parameter: systemEntityTypes.organization.entityTypeBaseUrl },
+            ],
+          },
+        ],
+      },
+      graphResolveDepths: zeroedGraphResolveDepths,
+      includeDrafts: true,
+      temporalAxes: currentTimeInstantTemporalAxes,
+    },
+  }).then((subgraph) => getRoots(subgraph));
```

Comment:
Why are these two different sequential queries? We could either call these simultaneously, or just call a single query with an `any` condition getting all entities that are users or orgs, and then filter the result for the users and orgs.

Reply:
I didn't really think about it much – I don't think we need to optimize the migration scripts until running them becomes a big part of start-up time (there are lots of property types we create sequentially where we could in parallel, for example). I've updated these to run in parallel in https://github.com/hashintel/hash/pull/3864/commits/d7121856688a3b25ca25f7646f3ff2c25ade48cd